### PR TITLE
javasrc2cpg: Update JavaDataflowFixture to accept extra flows

### PR DESCRIPTION
**Pull request by evilpan**
This extended the test suite to add custom semantic flows as said in https://github.com/joernio/joern/pull/2514.
But I can't managed to re-open the closed PR, so I just create a new one.

### CODE REVIEW

**Review by DavidBakerEffendi — APPROVED** [review]
Looks great, perhaps just another pair of eyes to approve, then we can merge cc: @johannescoetzee

File: joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/dataflow/SemanticTests.scala [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      List(
+        FlowSemantic("Test.sanitize:java.lang.String(java.lang.String)", List((0, 0), (1, 1))),
+        FlowSemantic("^ext\\.Library\\.taintReturn:.*", List((0, 0), (1, -1)), regex = true),
+        FlowSemantic("^ext\\.Library\\.taintNone:.*", List((0, 0), (1, 1)), regex = true),
+        FlowSemantic("^ext\\.Library\\.taint1to2:.*", List((1, 2)), regex = true)
+      )
+    ) {
+  behavior of "Dataflow through custom semantics"
+
+  // new Parser().parse("""
```
Comment (fabsx00) [inline joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/dataflow/SemanticTests.scala:23]: Can we get rid off this commented out code?

Comment (evilpan) [reply joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/dataflow/SemanticTests.scala:23]: Yeah sure, I'll add more tests and clean those comments by the way.

Comment (evilpan) [reply joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/dataflow/SemanticTests.scala:23]: See: https://github.com/joernio/joern/pull/2587

**Pull request merged**